test/dist_autograd_test.py

```diff
@@ -60,63 +78,141 @@ def test_autograd_context(self):
             ):
                 dist_autograd._retrieve_context(context_id)
 
-    @dist_init
-    def test_autograd_functions(self):
+    # Client sends tensors and then receives tensor result
+    def _verify_send_recv_functions_in_client(self, context_id, t1, t2, ret):
+        # Get send function.
+        ctx = dist_autograd._current_context()
+        self.assertEqual(context_id, ctx._context_id())
+        send_functions = ctx._send_functions()
+        self.assertEqual(1, len(send_functions))
+
+        # Retrieve the next functions in the graph.
+        next_funcs = list(send_functions.values())[0].next_functions
+        self.assertEqual(2, len(next_funcs))
+
+        # We should now hit t1 and t2 in the autograd graph.
+        self.assertEqual("torch::autograd::AccumulateGrad", next_funcs[0][0].name())
+        self.assertEqual(t1, next_funcs[0][0].variable)
+        self.assertEqual(0, next_funcs[0][1])
+        self.assertEqual("torch::autograd::AccumulateGrad", next_funcs[1][0].name())
+        self.assertEqual(t2, next_funcs[1][0].variable)
+        self.assertEqual(0, next_funcs[1][1])
+
+        # Test recv functions.
+        recv_functions = ctx._recv_functions()
+        self.assertEqual(1, len(recv_functions))
+        self.assertEqual(ret.grad_fn, list(recv_functions.values())[0])
+
+    # Host receives tensors and actually runs tensor operations, return tensor
```

Comment:
Could you please elaborate a bit more in the comments? What does `_in_tensor_run` mean? Isn't this function verifying the autograd graph structure?

Reply:
yes, the naming is bad, I will change it